src/common/procedure/src/procedure.rs

```diff
@@ -174,6 +216,59 @@ impl<T: Procedure + ?Sized> Procedure for Box<T> {
     fn lock_key(&self) -> LockKey {
         (**self).lock_key()
     }
+
+    fn poison_keys(&self) -> PoisonKeys {
+        (**self).poison_keys()
+    }
+}
+
+#[derive(Clone, Debug, PartialEq, Eq, Hash, PartialOrd, Ord, AsRefStr)]
+pub enum ResourceType {
```

Comment:
I don't think we need the `ResourceType` or something like that.

Just the string key is enough. We can define some functions to generate the `PoisonKey`:

```rust
fn table_poison_key(table_id: TableId) -> PoisonKey{
  PoisonKey(format!("table/{table_id}")
}
```

Reply:
Leave the user to define the unique poison key.

Reply:
resolved in [dc9e3b5](https://github.com/GreptimeTeam/greptimedb/pull/5822/commits/dc9e3b5cf3528d7a248e6fe7641687432a55a307)